posts/2021/09/hypothesis-array-api.ipynb

```diff
@@ -0,0 +1,940 @@
+      "Falsifying example: test_positive_arrays_have_incrementing_sums(\n",
+      "    x=Array([[False, False]], dtype=bool),\n",
+      ")\n",
+      "TypeError: only size-1 arrays can be converted to Python scalars\n",
+      "\n",
+      "The above exception was the direct cause of the following exception:\n",
+      "\n",
+      "Traceback (most recent call last):\n",
+      "  <cell>, line 12, in test_positive_arrays_have_incrementing_sums\n",
+      "    a = cumulative_sums(x)\n",
+      "  <cell>, line 13, in cumulative_sums\n",
+      "    result[0] = x[0]\n",
+      "  File \"<env>/numpy/array_api/_array_object.py\", line 657, in __setitem__\n",
+      "    self._array.__setitem__(key, asarray(value)._array)\n",
+      "ValueError: setting an array element with a sequence.\n",
+      "\n",
+      "Falsifying example: test_positive_arrays_have_incrementing_sums(\n",
+      "    x=Array([False, False], dtype=bool),\n",
+      ")\n",
+      "Traceback (most recent call last):\n",
+      "  <cell>, line 12, in test_positive_arrays_have_incrementing_sums\n",
+      "    a = cumulative_sums(x)\n",
+      "  <cell>, line 15, in cumulative_sums\n",
+      "    result[i] = result[i - 1] + x[i]\n",
+      "  File \"<env>/numpy/array_api/_array_object.py\", line 362, in __add__\n",
+      "    other = self._check_allowed_dtypes(other, \"numeric\", \"__add__\")\n",
+      "  File \"<env>/numpy/array_api/_array_object.py\", line 125, in _check_allowed_dtypes\n",
+      "    raise TypeError(f\"Only {dtype_category} dtypes are allowed in {op}\")\n",
+      "TypeError: Only numeric dtypes are allowed in __add__\n",
+      "=========================== short test summary info ============================\n",
+      "FAILED <cell>::test_positive_arrays_have_incrementing_sums - hypothes...\n",
+      "\u001b[31m\u001b[31m\u001b[1m1 failed\u001b[0m, \u001b[33m1 deselected\u001b[0m\u001b[31m in 0.49s\u001b[0m\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "from math import prod\n",
+    "from hypothesis import settings\n",
+    "\n",
+    "@given(\n",
+    "    xps.arrays(\n",
+    "        dtype=xps.scalar_dtypes(),\n",
+    "        shape=xps.array_shapes().filter(lambda s: prod(s) > 1),\n",
+    "        elements={\"min_value\": 0},\n",
+    "    )\n",
+    ")\n",
+    "def test_positive_arrays_have_incrementing_sums(x):\n",
+    "    a = cumulative_sums(x)\n",
+    "    assert nxp.all(a[1:] >= a[:-1])\n",
+    "    \n",
+    "ipytest.run(\"-k positive_arrays_have_incrementing_sums\", \"--hypothesis-seed=3\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73fd0ff6",
+   "metadata": {},
+   "source": [
+    "Again Hypothesis has proved our assumptions wrong,\n",
+    "and this time it's found two problems.\n",
+    "\n",
+    "Firstly, our `cumulative_sums()` method doesn't adjust for boolean arrays,\n",
+    "so we get an error when we blindly add two `bool` values together.\n",
```

Comment:
I would not use "blindly" here, but rather go with "inadvertently" or even remove the qualifier altogether.